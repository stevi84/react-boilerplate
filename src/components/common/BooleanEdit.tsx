import { Checkbox, FormControlLabel } from '@mui/material';
import { useField } from 'formik';

interface BooleanEditProps {
  id: string;
  name: string;
  label: string;
}

export const BooleanEdit = (props: BooleanEditProps) => {
  const { id, name, label } = props;

  const [field] = useField({ name, type: 'checkbox' });

  return (
    <FormControlLabel
      control={<Checkbox id={id} checked={field.checked} onChange={field.onChange} sx={{ margin: 1 }} />}
      label={label}
    />
  );
};
