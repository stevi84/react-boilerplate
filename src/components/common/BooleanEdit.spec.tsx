import { describe, it, expect, vi } from 'vitest';
import { BooleanEdit } from './BooleanEdit';
import { fireEvent, render, screen } from '@testing-library/react';

const { onChangeMock } = vi.hoisted(() => ({ onChangeMock: vi.fn() }));
vi.mock('formik', () => ({ useField: () => [{ onChange: onChangeMock, checked: true }] }));

describe('BooleanEdit', () => {
  it('should equal saved snapshot', () => {
    const tree = render(<BooleanEdit id="id" name="name" label="label" />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it('should trigger onChange', () => {
    render(<BooleanEdit id="id" name="name" label="label" />);
    expect(onChangeMock.mock.calls.length).toEqual(0);
    fireEvent.click(screen.getByTestId('CheckBoxIcon'));
    expect(onChangeMock.mock.calls.length).toEqual(1);
  });
});
