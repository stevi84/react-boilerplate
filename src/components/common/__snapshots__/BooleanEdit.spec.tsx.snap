// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BooleanEdit > should equal saved snapshot 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1m3cev1-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        checked=""
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        id="id"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
    >
      label
    </span>
  </label>
</DocumentFragment>
`;
